tools/sweeper_resolver/main.go

```diff
@@ -0,0 +1,293 @@
+package main
+
+import (
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/GoogleCloudPlatform/magic-modules/mmv1/api"
+	"github.com/GoogleCloudPlatform/magic-modules/tools/test-reader/reader"
+	"gopkg.in/yaml.v2"
+)
+
+// Custom logger that only prints when debug mode is enabled
+type debugLogger struct {
+	debug bool
+}
+
+func (l *debugLogger) Printf(format string, v ...interface{}) {
+	if l.debug {
+		log.Printf(format, v...)
+	}
+}
+
+type MetaData struct {
+	Resource       string `yaml:"resource"`
+	GenerationType string `yaml:"generation_type"`
+	SourceFile     string `yaml:"source_file,omitempty"`
+}
+
+func main() {
+	// Define command line flags
+	dirPath := flag.String("dir", "", "Path to services directory")
+	debug := flag.Bool("debug", false, "Enable debug logging")
+	flag.Parse()
+
+	// Initialize debug logger
+	logger := &debugLogger{debug: *debug}
+
+	// Configure logging format if debug is enabled
+	if *debug {
+		log.SetFlags(log.LstdFlags | log.Lshortfile)
+	}
+
+	logger.Printf("Starting program with directory path: %s", *dirPath)
+
+	if *dirPath == "" {
+		log.Fatal("Please provide a directory path using -dir flag")
+	}
+
+	allTests, errs := reader.ReadAllTests(*dirPath)
```

Comment:
Ideally we'd want this to be integrated in the CI pipeline, which is already reading tests in order to do missing test detection. It would be great to only read them once, then do all the relevant processing of that data (for missing tests, sweeper regions, and sweepable prefix detection), and post any suggested changes as inline comments (or comments on the file in cases where inline doesn't work for some reason).